app/src/app/components/sidebar/RecentMapsModal.tsx

```diff
@@ -0,0 +1,157 @@
+import { useMapStore } from "@/app/store/mapStore";
+import React from "react";
+import { Cross2Icon } from "@radix-ui/react-icons";
+import {
+  Button,
+  Flex,
+  Text,
+  Table,
+  Dialog,
+  Box,
+  TextField,
+  IconButton,
+} from "@radix-ui/themes";
+import { usePathname, useSearchParams, useRouter } from "next/navigation";
+import { DocumentObject } from "../../utils/api/apiHandlers";
+type NamedDocumentObject = DocumentObject & { name?: string };
+export const RecentMapsModal = () => {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const mapDocument = useMapStore((store) => store.mapDocument);
+  const userMaps = useMapStore((store) => store.userMaps);
+  const upcertUserMap = useMapStore((store) => store.upcertUserMap);
```

Comment:
PP: should be "upsert" instead of "upcert" throughout to avoid confusion later on